docs/tech-preview-blog-post.md

```diff
@@ -0,0 +1,130 @@
+These messages can dynamically include data values formatted
+(using information in the Common Locale Data Repository [CLDR])
+according to the needs of the language and culture of the end user.
+Such messages can be adjusted to meet the linguistic needs of each
+language and are designed to be translated easily and efficiently.
+
+Previously, software developers had to choose between many different
+APIs and templating languages to build user interface strings.
+These solutions did not always provide for the features of different
+human languages. Support was limited to specific platforms
+and these formats were not widely supported by translation tools,
+making translation and adaptation to specific cultures costly
+and time consuming.
+Most significantly, message formatting was limited to a small
+number of built-in formats.
+
+One of the challenges in adapting software to work for
+users with different languages and cultures is the need for **_dynamic messages_**.
+Whenever a user interface needs to present data as part of a larger string,
+that data needs to be formatted. 
+In many languages, including English, the message itself needs to be altered
+to make it grammatically correct.
+
+For example, if a message in English might read:
+
+> Your item had **1,023** views on **April 8, 2024**.
+
+The equivalent message in French might read:
+
+> Votre article a eu **1 023** vues le **8 avril 2024**.
+
+Or Japanese:
+
+> あなたのアイテムは **2024 年 4 月 8 日**に **1,023** 回閲覧されました。
+
+But even in English, there are grammatical variations required:
+
+> Your item had _no views_...
+> 
+> Your item had 1 _view_...
+> 
+> Your item had 1,043 _views_...
+
+Once messages have been created, they need to be translated into the various
+languages and adapted for the various cultures around the world.
+Previously there was not widely adopted standard
+and existing formats provided only rudimentary support for managing
+the variations needed by other languages,
+it could be difficult for translators to do their work effectively.
+
+For example, the same message shown above, needs a different set of variations
+in order to support Polish:
+
+> Twój przedmiot nie _ma_ żadnych _wyświetleń_.
+> 
+> Twój przedmiot _miał_ 1 _wyświetlenie_.
+> 
+> Twój przedmiot _miał_ 2 _wyświetlenia_.
+> 
+> Twój przedmiot _ma_ 5 _wyświetleń_.
+
+
+MessageFormat 2 makes it easy to write messages like this
+without developering needing to know about such language variation.
+In fact, developers don't need to learn about any of the language
+and formatting variations needed by languages other than their own
+nor write code that manipulates formatting.
+
+MessageFormat 2 messages can be simple strings:
+```
+    Hello, world!
+```
+
+A message can also include _placeholders_ that are replaced by user-provided values:
+```
+    Hello {$user}!
+```
+
+The user-provided values can be transformed or formatted using functions:
+```
+    Today is {$date :datetime}
```

Comment:
With a bare `:datetime`, this would format as "Today is 4/5/24, 4:19 PM", which is probably not the intent?

```suggestion
    Today is {$date :date}
```